ui/src/shared/components/LayoutCell.js

```diff
@@ -54,7 +75,13 @@ class LayoutCell extends Component {
         />
         <div className="dash-graph--container">
           {queries.length
-            ? children
+            ? React.Children.map(children, child => {
+                if (child && child.props && child.props.autoRefresh) {
+                  return React.cloneElement(child, {
+                    grabDataForDownload: this.grabDataForDownload,
+                  })
+                }
+              })
```

Comment:
Compose those components!

Reply:
Yeah, is this necessary? Can't this just be a regular component instantiation?